tests/Database/DatabaseEloquentBelongsToManyTest.php

```diff
@@ -116,7 +116,7 @@ public function testModelsAreProperlyMatchedToParents()
         $this->assertEquals(2, $models[1]->foo[0]->pivot->user_id);
         $this->assertEquals(2, $models[1]->foo[1]->pivot->user_id);
         $this->assertEquals(2, count($models[1]->foo));
-        $this->assertEquals(0, count($models[2]->foo));
+        $this->assertEquals(0, count((array) $models[2]->foo));
```

Comment:
So to me, these tests prove we are only hiding the issue?

The real "solution" would be to allow the result to always be "countable"? Otherwise you could no longer use count() safely on a model result?